### PR TITLE
Modifications to allow further Jackson based media type support

### PR DESCRIPTION
**Pull request by jameskleeh**

### CODE REVIEW

File: xml/src/main/java/io/micronaut/xml/codec/XmlMediaTypeCodec.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+public class XmlMediaTypeCodec extends JacksonMediaTypeCodec {
+
+    public static final String CONFIGURATION_QUALIFIER = "xml";
+
+    /**
+     * @param xmlMapper                Object mapper for xml. If null, retrieved from beanContext
+     * @param applicationConfiguration The common application configurations
+     * @param codecConfiguration       The configuration for the codec
+     */
+    public XmlMediaTypeCodec(@Named(CONFIGURATION_QUALIFIER) ObjectMapper xmlMapper,
```
Comment (svishnyakoff) [inline xml/src/main/java/io/micronaut/xml/codec/XmlMediaTypeCodec.java]: So developers are able to customize ObjectMapper for json, but not for xml? if both media type are processed with Jackson IMHO `@JackongFeatures` should be applicable for both media types

File: xml/src/main/java/io/micronaut/xml/server/XmlContentProcessor.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    @Override
+    protected void onUpstreamMessage(ByteBufHolder message) {
+        ByteBuf content = message.content();
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Buffer xml bytes of size {}", content.readableBytes());
+        }
+
+        try {
+            content.readBytes(byteArrayStream, content.readableBytes());
```
Comment (svishnyakoff) [inline xml/src/main/java/io/micronaut/xml/server/XmlContentProcessor.java]: 👍

File: xml/src/main/java/io/micronaut/xml/codec/XmlMediaTypeCodec.java (hunk 1 above)
Comment (jameskleeh) [reply xml/src/main/java/io/micronaut/xml/codec/XmlMediaTypeCodec.java]: Where are you getting the impression that it isn't?

Comment (svishnyakoff) [reply xml/src/main/java/io/micronaut/xml/codec/XmlMediaTypeCodec.java]: I meant customization using `@JacksonFeatures` annotation that is applied on the client and only processed within `HttpClientIntroductionAdvice`. We use annotation data to customize objectMapper for `json` media type,  and not doing same for `xml`

**Comment by jameskleeh** [comment]
This PR accomplishes that. You should either look at the client
introduction advice code or try it yourself

File: xml/src/main/java/io/micronaut/xml/codec/XmlMediaTypeCodec.java (hunk 1 above)
Comment (svishnyakoff) [reply xml/src/main/java/io/micronaut/xml/codec/XmlMediaTypeCodec.java]: ahh see now, false alarm :)

**Comment by graemerocher** [comment]
In general this looks ok to me. A few of changes:

* Rename `io.micronaut.xml` to `io.micronaut.jackson.xml`. We don't know in the future whether we want to support other XML binders, like JAX-B etc.
* Rename the module `xml` -> `jackson-xml`
* Can we move the module into a new subproject under `micronaut-projects`? Is there some requirement for this to be in core?

**Comment by graemerocher** [comment]
@jameskleeh will you be making these changes so this can go into 1.3 M2?

**Comment by jameskleeh** [comment]
Yes I will today

**Pull request merged**